cancat/uds.py

```diff
@@ -314,24 +338,49 @@ def RequestUpload(self, addr, length, data_format = 0x00, addr_format = 0x44):
     def EcuReset(self, rst_type=0x1):
         return self._do_Function(SVC_ECU_RESET, subfunc=rst_type)
 
-    def ClearDiagnosticInformation(self):
-        pass
-    def ReadDTCInfomation(self):
-        pass
-    def ReadDataByPeriodicIdentifier(self, pdid):
-        pass
+    def ClearDiagnosticInformation(self, groupOfDTC):
+        msg = self._do_Function(SVC_CLEAR_DIAGNOSTICS_INFORMATION,data =struct.pack('>HB', groupOfDTC >> 8, groupOfDTC & 0xFF), service=0x54)
+        #msg = self._do_Function(SVC_CLEAR_DIAGNOSTICS_INFORMATION,data =groupOfDTC.decode('hex'), service=0x54)
+        return msg
+    def ReadDTCInformation(self, subfunc,data=None):
+        if data != None:
+            msg = self._do_Function(SVC_READ_DTC_INFORMATION,subfunc =subfunc, data= struct.pack('>H', data)[1:], service=0x59)
+        else:
+            msg = self._do_Function(SVC_READ_DTC_INFORMATION,subfunc =subfunc, service=0x59)
+        return msg
+    def ReadDataByPeriodicIdentifier(self, tmode, pdid):
+        msg = self._do_Function(SVC_READ_DATA_BY_PERIODIC_IDENTIFIER,subfunc =tmode, data= struct.pack('>B', pdid) , service=0x6a)
+        return msg
     def DynamicallyDefineDataIdentifier(self):
         pass
-    def InputOutputControlByIdentifier(self, iodid):
-        pass
-    def RoutineControl(self, rid):
-        pass
-    def TransferData(self, did):
-        pass
-    def RequestTransferExit(self):
-        pass
-    def ControlDTCSetting(self):
-        pass
+    def InputOutputControlByIdentifier(self, iodid, controlOptionRecord):
+        #msg = self._do_Function(SVC_INPUT_OUTPUT_CONTROL_BY_IDENTIFIER,subfunc = iodid, data= data, service=0x6f)
+        #return msg
+    def RoutineControl(self, routineControlType, rid, routineCOR=None):
```

Comment:
What does COR stand for? Might want to add a comment above this function explaining which parameter means what